moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -0,0 +1,355 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2016 Shaun Edwards
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#pragma once
+
+#include <vector>
+#include <map>
+#include <string>
+#include <memory>
+#include <mutex>
+
+#include <eigen3/Eigen/Geometry>
+
+#include <ros/ros.h>
+
+#include <moveit/planning_scene/planning_scene.h>
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit_visual_tools/moveit_visual_tools.h>
+
+#include <tf2_ros/transform_listener.h>
+
+#include <trajectory_msgs/JointTrajectoryPoint.h>
+#include <actionlib/client/simple_action_client.h>
+#include <control_msgs/FollowJointTrajectoryAction.h>
+#include <sensor_msgs/JointState.h>
+
+
+
+namespace moveit_simple
+{
+
+class TrajectoryPoint;
+class JointTrajectoryPoint;
+class CartTrajectoryPoint;
+class Robot;
+typedef std::vector<std::unique_ptr<TrajectoryPoint> > Trajectory;
+
+
+/**
+ * @brief Robot is a wrapper around standard MoveIt objects.  It makes multiple
+ assumptions about the type and complexity of the robot.  Assumptions are:
+ - single arm manipulator (one joint group)
```

Comment:
I'm open to changing the name, especially if we pull this into MoveIt.  Let's wait to see what functionality ultimately shakes out.  I'm the only one depending on this code at the moment, so it will only cause myself pain to change the name.